HopsanGUI/Dialogs/ComponentPropertiesDialog3.cpp

```diff
@@ -259,6 +260,14 @@ void ComponentPropertiesDialog3::applyPressed()
     setVariableValues();
 }
 
+void ComponentPropertiesDialog3::openSourceCode()
+{
+    auto& appearance = gpLibraryHandler->getModelObjectAppearancePtr(mpModelObject->getTypeName());
+    QString basePath = appearance->getBasePath();
+    QString fileName = appearance->getSourceCodeFile();
+    gpModelHandler->loadTextFile(basePath+fileName);
```

Comment:
I dont think that we can assume that basepath includes a trailing /, in my opinion it should not. So a / must br added, but only if basePath is not empty... we need some  clever helpfunction for this. But I assumt that the currnt  code works for some reason, maybe basepath has a trailing /

Reply:
Perhaps all functions returning a path should be responsible of adding a trailing slash if it is missing? I will just add an if statement to solve it for now.

Reply:
No, this is how it was previously solved, but then sometimes you do not want a trailing slash, I have had lots of problems with this sort of thing. The correct way is to never include a trailing slash, and then use a helpfunction similar to pythons path.join(part1,part2) that is clever and adds / as needed. But we can add that later